### PR TITLE
Kehyde/date range picker

### PR DESCRIPTION
**Pull request by futurethang**
New Component Date Range Picker

## Description
The component builds on top of the existing Date Picker API and adds features and style to generate a 2 date range.

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -1,10 +0,0 @@
-coverage:
```
Comment (yananym) [inline codecov.yml:1]: ghhhmmm i needs that file buddy :)

File: web-components/src/components/date-range-picker/DateRangePicker.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { customElementWithCheck } from "@/mixins/CustomElementCheck";
```
Comment (yananym) [inline web-components/src/components/date-range-picker/DateRangePicker.ts:1]: please don't forget the disclaimer in all new files

File: web-components/src/components/datepicker/scss/datepicker.scss [hunk 3]
```diff
@@ -74,12 +73,20 @@
...
+        left: 7%;
+        position: absolute;
+        top: 4%;
+        width: 1.8rem;
+      }
     }
 
     &--focus::part(button) {
-      background: $md-orange-50;
+      background: #dedede;
```
Comment (yananym) [inline web-components/src/components/datepicker/scss/datepicker.scss:89]: really? jut a hex value like that?

File: web-components/src/index.ts [hunk 4]
```diff
@@ -24,7 +25,8 @@ export { Chip } from "./components/chip/Chip";
 export { Coachmark } from "./components/coachmark/Coachmark";
 export { CodeEditor } from "./components/code-editor/CodeEditor";
 export { ComboBox } from "./components/combobox/ComboBox";
-export { CompositeAvatar } from "./components/avatar/CompositeAvatar";
+export { DateRangePicker } from "./components/date-range-picker/DateRangePicker";
```
Comment (yananym) [inline web-components/src/index.ts:28]: don't forget to add entry point to webpack

File: web-components/src/[sandbox]/examples/index.ts [hunk 5]
```diff
@@ -13,13 +13,15 @@ export { chipTemplate } from "./chip";
 export { coachTemplate } from "./coachmark";
 export { codeEditorTemplate } from "./code-editor";
 export { comboBoxTemplate } from "./combobox";
-export { datePickerTemplate } from "./datepicker";
+export { dateRangePickerTemplate } from "./date-range-picker";
 export { dateTimePickerTemplate } from "./date-time-picker";
+export { datePickerTemplate } from "./datepicker";
```
Comment (marinasmah) [inline web-components/src/[sandbox]/examples/index.ts:18]: @futurethang Plz fix double export "datePickerTemplate"

**Comment by marinasmah** [comment]
@futurethang plz check styles. They are different from [Design](https://www.figma.com/file/wDbhcTS6fYP9wFyJ7VnE0r/Lumos-Components?node-id=11076%3A96924)

File: codecov.yml (hunk 1 above)
Comment (futurethang) [reply codecov.yml:1]: very strange . . . I'm not sure how that happenned but got it back in there

File: web-components/src/components/date-range-picker/DateRangePicker.ts (hunk 2 above)
Comment (futurethang) [reply web-components/src/components/date-range-picker/DateRangePicker.ts:1]: @yananym just to be 100% clear, same goes for all scss, tokens, test files?

**Comment by futurethang** [comment]
> @futurethang plz check styles. They are different from [Design](https://www.figma.com/file/wDbhcTS6fYP9wFyJ7VnE0r/Lumos-Components?node-id=11076%3A96924)

Correct, I have been having discussions with Mocha about opting for a single or double up window, and which is most needed at the moment.

File: web-components/src/components/date-range-picker/DateRangePicker.ts (hunk 2 above)
Comment (yananym) [reply web-components/src/components/date-range-picker/DateRangePicker.ts:9]: right

File: web-components/src/[sandbox]/sandbox.ts [hunk 6]
```diff
@@ -276,9 +284,7 @@ export class Sandbox extends LitElement {
 
           <div class="container" aria-label="md-form">
             <h2>md-form</h2>
-            <sass-stats component="form">
```
Comment (yananym) [inline web-components/src/[sandbox]/sandbox.ts:279]: why did you remove sass stats here?

File: web-components/src/components/date-range-picker/DateRangePicker.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      });
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      this.removeEventListener("date-selection-change", () => {});
+    }
+
+    // TO DO : Need to force a re-render on the first date selection so that the DatePickerDay component renders the class md-datepicker__day--start-date and has filled-in black style
```
Comment (yananym) [inline web-components/src/components/date-range-picker/DateRangePicker.ts:35]: todo comments?

File: web-components/src/components/date-range-picker/DateRangePicker.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+    // TODO: Find out why Key nav freezes
```
Comment (yananym) [inline web-components/src/components/date-range-picker/DateRangePicker.ts:37]: can this be cleaned up?

File: web-components/src/[sandbox]/sandbox.ts (hunk 6 above)
Comment (futurethang) [reply web-components/src/[sandbox]/sandbox.ts:279]: there are no styles for form, it's all derived from <input>, and therefore no stats get compiled for display.

File: web-components/src/components/date-range-picker/DateRangePicker.ts (hunk 7 above)
Comment (futurethang) [reply web-components/src/components/date-range-picker/DateRangePicker.ts:35]: my bad, thanks for pointing out

**Review by yananym — APPROVED** [review]

**Pull request merged**